nomos-services/chain/chain-service/src/lib.rs

```diff
@@ -1121,6 +1158,49 @@ where
             if let Err(e) = lib_broadcaster.send(lib_update) {
                 error!("Could not notify LIB update to services: {e}");
             }
+
+            let previous_session_numbers = match cryptarchia.active_sessions_numbers(&prev_lib) {
+                Ok(session_numbers) => session_numbers,
+                Err(e) => {
+                    warn!("Error getting previous session numbers: {e}");
+                    ServiceType::iter().map(|s| (s, 0)).collect()
+                }
+            };
+
+            if let Ok(sessions) = cryptarchia.active_sessions_numbers(&new_lib) {
```

Comment:
```suggestion
            let Ok(sessions) = cryptarchia.active_sessions_numbers(&new_lib) else {
```

Reply:
Updated with your suggestion to return an error when active session numbers cant be recieved (should never happen) https://github.com/logos-co/nomos/pull/1786/commits/0c9b3fc1e9d75296e3a12ca5341e335439d5cad3

Reply:
My suggestion was simply about replacing the very big if block with an `if-let-else` that would remove one layer of indentation, else `continue`ing. Sorry I should have made this clear. If the error should never happen, then there is no need to throw one. Either we panic or log a `error` message and move on. Again, sorry that I did not provide more context about my suggestion.